### ISSUE TITLE
Need to re-compile model after changing the config of a model layer ?

### ISSUE THREAD

**Issue by jasonTuZq** [fchollet/keras#6315]
Hi, do i need to re-compile model after changing the config of a model layer ?
For example, i have a layer Dropout(0.5), which is in the third layer, when i change its config like this;
     cfg = model.layers[3].get_config()
     cfg['p']=0.2
     model.layers[3] = model.layers[3].from_config(cfg)
(ps: Is this a proper way to change the config of a model layer?)
do i need to need to re-compile the model ?

Thanks !

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
